lib/rules/media-feature-name-value-no-unknown/index.js

```diff
@@ -0,0 +1,318 @@
+		 * @param {State} state
+		 * @param {Array<import('@csstools/css-parser-algorithms').ComponentValue>} componentValues
+		 * @param {Reporter} reporter
+		 * @returns {void}
+		 */
+		function checkListOfComponentValues(state, componentValues, reporter) {
+			const supportedValueTypes = mediaFeatureNameAllowedValueTypes.get(state.mediaFeatureName);
+
+			if (
+				supportedValueTypes &&
+				supportedValueTypes.has('ratio') &&
+				matchesRatioExactly(componentValues) !== -1
+			) {
+				return;
+			}
+
+			// An invalid aspect ratio or a media feature that doesn't support aspect ratios.
+			reporter(
+				state,
+				componentValues.map((x) => x.toString()).join(''),
+				...startAndEndIndex(componentValues),
+			);
+		}
+
+		/**
+		 * @param {State} state
+		 * @param {import('@csstools/media-query-list-parser').MediaFeatureValue} valueNode
+		 * @param {Reporter} reporter
+		 * @returns {void}
+		 */
+		function checkMediaFeatureValue(state, valueNode, reporter) {
+			if (isTokenNode(valueNode.value)) {
+				checkSingleToken(state, valueNode.value.value, reporter);
+
+				return;
+			}
+
+			if (isFunctionNode(valueNode.value)) {
+				checkFunction(state, valueNode.value, reporter);
+
+				return;
+			}
+
+			if (Array.isArray(valueNode.value)) {
+				checkListOfComponentValues(state, valueNode.value, reporter);
+			}
+		}
+
+		root.walkAtRules(/^media$/i, (atRule) => {
+			/**
+			 * @type {Reporter}
+			 */
+			const reporter = (state, valuePart, start, end) => {
+				const atRuleParamIndexValue = atRuleParamIndex(atRule);
+
+				report({
+					message: messages.rejected,
+					messageArgs: [state.mediaFeatureNameRaw, valuePart],
+					index: atRuleParamIndexValue + start,
+					endIndex: atRuleParamIndexValue + end,
+					node: atRule,
+					ruleName,
+					result,
+				});
+			};
+
+			/** @type {State} */
+			const state = {
+				mediaFeatureName: '',
+				mediaFeatureNameRaw: '',
+			};
+
+			parseMediaQuery(atRule).forEach((mediaQuery) => {
+				mediaQuery.walk((entry) => {
+					if (!entry.state) return;
+
+					if (isMediaFeature(entry.node)) {
+						const mediaFeatureNameRaw = entry.node.getName();
+						let mediaFeatureName = vendor.unprefixed(mediaFeatureNameRaw.toLowerCase());
+
+						// Unknown media feature names are handled by "media-feature-name-no-unknown".
+						if (!mediaFeatureNames.has(mediaFeatureName)) return;
+
+						if (HAS_MIN_MAX_PREFIX.test(mediaFeatureName)) {
+							mediaFeatureName = mediaFeatureName.slice(4);
+						}
+
+						entry.state.mediaFeatureName = mediaFeatureName;
+						entry.state.mediaFeatureNameRaw = mediaFeatureNameRaw;
+
+						return;
+					}
+
+					if (!entry.state.mediaFeatureName || !entry.state.mediaFeatureNameRaw) return;
+
+					if (isMediaFeatureValue(entry.node)) {
+						checkMediaFeatureValue(entry.state, entry.node, reporter);
+					}
+				}, state);
+			});
+		});
+	};
+};
+
+rule.ruleName = ruleName;
+rule.messages = messages;
+rule.meta = meta;
+module.exports = rule;
```

Comment:
[suggest] Can you please move the lines about exporting to the bottom of the file to easily find the exporting?